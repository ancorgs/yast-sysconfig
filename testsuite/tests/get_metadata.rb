# encoding: utf-8

# Sysconfig module - testsuite
#
# get_metadata tests
#
# testedfiles: Sysconfig.ycp
#
# $Id$
#
module Yast
  class GetMetadataClient < Client
    def main
      Yast.import "Testsuite"
      Yast.import "Pkg" # override packamanager
      Yast.import "Sysconfig"

      Testsuite.Test(lambda { Sysconfig.get_metadata(nil) }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { Sysconfig.get_metadata("") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { Sysconfig.get_metadata(" ## Type: ip  ") }, [
        {},
        {},
        {}
      ], nil)

      Testsuite.Test(lambda { Sysconfig.get_metadata("## Type: ip  ") }, [
        {},
        {},
        {}
      ], nil)
      Testsuite.Test(lambda { Sysconfig.get_metadata("##Type: ip") }, [
        {},
        {},
        {}
      ], nil)
      Testsuite.Test(lambda do
        Sysconfig.get_metadata("## Type: sdf  \n## Default: sdf")
      end, [
        {},
        {},
        {}
      ], nil)
      Testsuite.Test(lambda do
        Sysconfig.get_metadata("\n\n## Type\n   \n##Def\n  ##Path")
      end, [
        {},
        {},
        {}
      ], nil)

      nil
    end
  end
end

Yast::GetMetadataClient.new.main
