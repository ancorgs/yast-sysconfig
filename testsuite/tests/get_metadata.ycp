// Sysconfig module - testsuite
//
// get_metadata tests
//
// $Id$
//

{

include "testsuite.ycp";

import "Pkg";   // override packamanager

import "Mode";		// mode is imported in Product
Mode::config = true;	// package manager is initialized in Product constructor when Mode::config is false !!
import "Sysconfig";

TEST( ``(Sysconfig::get_metadata(nil) ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::get_metadata("") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::get_metadata(" ## Type: ip  ") ), [ $[], $[], $[] ], nil );

TEST( ``(Sysconfig::get_metadata("## Type: ip  ") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::get_metadata("##Type: ip") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::get_metadata("## Type: sdf  \n## Default: sdf") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::get_metadata("\n\n## Type\n   \n##Def\n  ##Path") ), [ $[], $[], $[] ], nil );

}

