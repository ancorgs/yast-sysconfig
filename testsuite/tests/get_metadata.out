Return	[]
Return	[]
Return	[]
Return	["## Type: ip  "]
Return	["##Type: ip"]
Return	["## Type: sdf  ", "## Default: sdf"]
Return	["## Type", "##Def"]
